mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -1293,28 +1293,78 @@ LogicalResult ModuleImport::convertInstruction(llvm::Instruction *inst) {
     if (failed(convertCallTypeAndOperands(invokeInst, types, operands)))
       return failure();
 
-    SmallVector<Value> normalArgs, unwindArgs;
-    (void)convertBranchArgs(invokeInst, invokeInst->getNormalDest(),
-                            normalArgs);
-    (void)convertBranchArgs(invokeInst, invokeInst->getUnwindDest(),
-                            unwindArgs);
+    // Check whether the invoke result is an argument to the normal destination
+    // block.
+    std::size_t i = 0;
+    std::optional<std::size_t> invokeBlockArgPos;
+    for (auto inst = invokeInst->getNormalDest()->begin();
+         !invokeBlockArgPos && isa<llvm::PHINode>(inst); ++inst, ++i) {
+      auto *phiInst = cast<llvm::PHINode>(&*inst);
+      llvm::Value *value =
+          phiInst->getIncomingValueForBlock(invokeInst->getParent());
+      if (value == invokeInst)
+        invokeBlockArgPos = i;
+    }
```

Comment:
You do not use the index, so suggesting something like:

```suggestion
    bool invokeResultUsedInPhi = false;
    for (const auto* phiInst : invokeInst->getNormalDest()->phis()) {
      if (phiInst->getIncomingValueForBlock(invokeInst->getParent()) == invokeInst) {
        invokeResultUsedInPhi = true;
        break;
      }
    }
```